langchain/src/tools/gmail/base.ts

```diff
@@ -21,46 +45,23 @@ export abstract class GmailBaseTool extends Tool {
   constructor(fields?: Partial<GmailBaseToolParams>) {
     super(...arguments);
 
-    const defaultCredentials = {
-      clientEmail: getEnvironmentVariable("GMAIL_CLIENT_EMAIL"),
-      privateKey: getEnvironmentVariable("GMAIL_PRIVATE_KEY"),
-      keyfile: getEnvironmentVariable("GMAIL_KEYFILE"),
-    };
-
-    const credentials = {
-      ...defaultCredentials,
-      ...(fields?.credentials ?? {}),
-    };
-
-    const scopes = fields?.scopes || ["https://mail.google.com/"];
-
-    if (!credentials) {
-      throw new Error("Missing credentials to authenticate to Gmail");
-    }
-
-    if (!credentials.clientEmail) {
-      throw new Error("Missing GMAIL_CLIENT_EMAIL to interact with Gmail");
-    }
-
-    if (!credentials.privateKey && !credentials.keyfile) {
-      throw new Error(
-        "Missing GMAIL_PRIVATE_KEY or GMAIL_KEYFILE to interact with Gmail"
-      );
-    }
+    const { credentials, scopes } = GmailBaseToolParamsSchema.parse(
+      fields ?? {}
```

Comment:
It does throw an error. It's because `GmailBaseToolParamsSchema` is "required" technically. I just played around with it a bit and it seems like an alternate solution is to change the schema to
```typescript
const GmailBaseToolParamsSchema = z
  .object({
    credentials: CredentialsSchema.default({}),
    scopes: z.array(z.string()).default(["https://mail.google.com/"]),
  })
  .default({}); // Adding this line to allow it to fallback to nothing 
// (which I think makes it fallback to the default environment variables)
```
I'll make that change now cuz it's cleaner than having to do `fields ?? {}` everytime